module/VuFind/src/VuFind/Auth/SimulatedSSO.php

```diff
@@ -71,10 +91,31 @@ public function __construct($url)
     public function authenticate($request)
     {
         // If we made it this far, we should log in the user!
-        $user = $this->getUserTable()->getByUsername('fakeuser1');
-        $user->firstname = 'Test';
-        $user->lastname = 'User';
-        $user->email = 'fake@example.com';
+        $username = $this->simulatedSSOConfig['General']['username'] ?? 'fakeuser1';
+        if (!$username) {
+            throw new AuthException('Simulated failure');
+        }
+        $user = $this->getUserTable()->getByUsername($username);
+        $attribs = array_merge(
```

Comment:
Those three attributes are generally expected, but you raise the good point that it is useful to be able to test what happens when they are absent. I've refined the configuration a bit so that defaults are provided in a backward-compatible, convenient way, but it's also possible to entirely disable attributes, or to test with a different basic set of values (without having any defaults forced in). See d494014.